proof/refine/ARM/orphanage/Orphanage.thy

```diff
@@ -699,12 +699,12 @@ lemma chooseThread_no_orphans [wp]:
   unfolding chooseThread_def Let_def
   supply if_split[split del]
   apply (simp only: return_bind, simp)
-  apply (intro hoare_seq_ext[OF _ stateAssert_sp])
-  apply (rule hoare_seq_ext[where B="\<lambda>rv s. ?PRE s \<and> ksReadyQueues_asrt s \<and> ready_qs_runnable s
+  apply (intro bind_wp[OF _ stateAssert_sp])
+  apply (rule bind_wp[where Q'="\<lambda>rv s. ?PRE s \<and> ksReadyQueues_asrt s \<and> ready_qs_runnable s
                                             \<and> rv = ksCurDomain s"])
-   apply (rule_tac B="\<lambda>rv s. ?PRE s \<and> ksReadyQueues_asrt s \<and> ready_qs_runnable s
+   apply (rule_tac Q'="\<lambda>rv s. ?PRE s \<and> ksReadyQueues_asrt s \<and> ready_qs_runnable s
                              \<and> curdom = ksCurDomain s \<and> rv = ksReadyQueuesL1Bitmap s curdom"
-                in hoare_seq_ext)
+                in bind_wp)
```

Comment:
This is an example of the second category of proof changes from the PR description; the conditions in the various `bind_wp` rules are now `P`, `Q'` and `Q`, instead of `A`, `B` and `C`.